From 0dc9f830a2499013ea1a2d994b1bfdfb0a19e815 Mon Sep 17 00:00:00 2001
From: Hao Zheng <haoz@marvell.com>
Date: Wed, 30 Jan 2019 15:37:28 -0800
Subject: [PATCH 0105/1921] octeontx2-af: NPC KPU profile fix

This patch fixes a parser error when KPU parses incoming IPSec
ESP and AH packets

Change-Id: Ide4772965f758df8eb743587e6b02f9057c67b64
Signed-off-by: Hao Zheng <haoz@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index dbf19b96a96e..06b3b3dc6811 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -8620,7 +8620,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 0,
+		0, 0, 0, 2, 0,
 		NPC_S_KPU8_ESP, 20, 1,
 		NPC_LID_LC, NPC_LT_LC_IP,
 		0,
@@ -8628,7 +8628,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 0,
+		0, 0, 0, 2, 0,
 		NPC_S_KPU8_AH, 20, 1,
 		NPC_LID_LC, NPC_LT_LC_IP,
 		0,
@@ -8724,7 +8724,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 0,
+		0, 0, 0, 2, 0,
 		NPC_S_KPU8_ESP, 0, 1,
 		NPC_LID_LC, NPC_LT_LC_IP_OPT,
 		0,
@@ -8732,7 +8732,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 0,
+		0, 0, 0, 2, 0,
 		NPC_S_KPU8_AH, 0, 1,
 		NPC_LID_LC, NPC_LT_LC_IP_OPT,
 		0,
-- 
2.31.1

